packages/router-core/src/router.ts

```diff
@@ -2029,20 +2029,35 @@ export class RouterCore<
    *
    * @link https://tanstack.com/router/latest/docs/framework/react/api/router/NavigateOptionsType
    */
-  navigate: NavigateFn = async ({ to, reloadDocument, href, ...rest }) => {
-    if (!reloadDocument && href) {
+  navigate: NavigateFn = async ({
```

Comment:
can we add the publicHref somehow else as a prop here so we can use it but dont show it in the public API?